packages/rollup-dev-tools/src/tools/transpiler/transpiler.ts

```diff
@@ -248,19 +256,55 @@ export class TranspilerImpl implements Transpiler {
       errors
     )
 
-    return {
-      succeeded: true,
-      bytecode: bytecodeToBuffer(finalTranspiledBytecode),
+    if (errors.length > 0) {
+      return {
+        succeeded: false,
+        errors,
+      }
+    } else {
+      return {
+        succeeded: true,
+        bytecode: bytecodeToBuffer(finalTranspiledBytecode),
+      }
     }
   }
 
+  private splitCodeAndConstants(
+    bytecodeWithTaggedConstants: EVMBytecode
+  ): EVMBytecode[] {
+    const pushConstantOffsetOperations = bytecodeWithTaggedConstants.filter(
+      (val: EVMOpcodeAndBytes): boolean => {
+        return isTaggedWithReason(val, [OpcodeTagReason.IS_CONSTANT_OFFSET])
+      }
+    )
+    const allConstants: Buffer[] = pushConstantOffsetOperations.map(
+      (val: EVMOpcodeAndBytes): Buffer => {
+        return val.tag.metadata
+      }
+    )
+    const bytecodeBuf: Buffer = bytecodeToBuffer(bytecodeWithTaggedConstants)
+    const constantStarts: number[] = allConstants.map(
+      (theConst: Buffer): number => {
+        return bytecodeBuf.indexOf(theConst)
+      }
+    )
+    const lowestStartPC: number = Math.min(...constantStarts)
+    const lowestStartIndexInBytecode: number = bufferToBytecode(
+      bytecodeBuf.slice(0, lowestStartPC)
+    ).length
+
+    return [
```

Comment:
So after the first constant starts in bytecode, there's only constants/unreachable bytecode following it? So this is finding the first constant, removing all the extra bytecode after it, then transpiling the remaining bytecode, then adding that extra bytecode back in unchanged?

Reply:
Yep, that's right! Or at least, it's what I've seen in all cases of constants I've looked at.  I was working on a more robust solution which would allow constants to be placed arbitrarily in the code, but it became a big headache and I realized this much simpler solution.